### PR TITLE
#2 GET /api/topics

### PR DESCRIPTION
**Pull request by dentednerd**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  describe('GET', () => {
+    test('200: returns an array of topic objects with slug and description', async () => {
+      const { body: { topics } } = await request(app).get('/api/topics').expect(200);
+        topics.forEach((topic) => {
+        expect(topic.slug).toEqual(expect.any(String));
+        expect(topic.description).toEqual(expect.any(String));
+      });
+    });
+
+    test('404: returns "Not found" for an incorrect route', async () => {
```
Comment (rogersop) [inline __tests__/app.test.js:20]: I would move this test outside of your `/api/topics` describe block (maybe just into a root one `/`?) as it is not specific to that topics path

File: models/topics.models.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  const queryStr = `
+    SELECT *
+    FROM topics;
+  `;
+
+  const topics = await db
+    .query(queryStr)
+    .then(({ rows }) => rows);
+
+  if (!topics || !topics.length) {
```
Comment (rogersop) [inline models/topics.models.js:13]: This feels a bit preemptive, as you do not have a test (or really a way of testing for) this. 

For this I think that either:
- the topics table exists, but there are no topics, so we get an empty array back and send a 200 response (which is fine), or;
- the topics table doesn't exist yet, and there are bigger problems to worry about than sending a 404 'Topics not found' response

File: models/topics.models.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+exports.fetchAllTopics = async () => {
+  const queryStr = `
+    SELECT *
+    FROM topics;
+  `;
+
+  const topics = await db
+    .query(queryStr)
+    .then(({ rows }) => rows);
```
Comment (rogersop) [inline models/topics.models.js:11]: See comment in the next pull request about this `.then` being a bit unnecessary.

File: __tests__/app.test.js (hunk 1 above)
Comment (dentednerd) [reply __tests__/app.test.js:20]: Good point!

**Pull request merged**